src/hooks.ts

```diff
@@ -201,38 +262,60 @@ export const useExperiment: UseExperiment = (experimentKey, options = {}, overri
  *       ClientReady and DidTimeout provide signals to handle this scenario.
  */
 export const useFeature: UseFeature = (featureKey, options = {}, overrides = {}) => {
-  const { isServerSide, optimizely, timeout } = useContext(OptimizelyContext);
+  const { optimizely, isServerSide } = useContext(OptimizelyContext);
   if (!optimizely) {
     throw new Error('optimizely prop must be supplied via a parent <OptimizelyProvider>');
   }
 
-  // Helper function to return the current values for isEnabled and variables.
-  const getCurrentValues = useCallback<() => FeatureDecisionValues>(
+  const overrideAttrs = useCompareAttrsMemoize(overrides.overrideAttributes);
+  const getCurrentDecision: () => FeatureDecisionValues = useCallback(
     () => ({
-      isEnabled: optimizely.isFeatureEnabled(featureKey, overrides.overrideUserId, overrides.overrideAttributes),
-      variables: optimizely.getFeatureVariables(featureKey, overrides.overrideUserId, overrides.overrideAttributes),
+      isEnabled: optimizely.isFeatureEnabled(featureKey, overrides.overrideUserId, overrideAttrs),
+      variables: optimizely.getFeatureVariables(featureKey, overrides.overrideUserId, overrideAttrs),
     }),
-    [featureKey, overrides]
+    [optimizely, featureKey, overrides.overrideUserId, overrideAttrs]
   );
 
-  // Set the initial state immediately serverSide
-  const [state, setState] = useState<UseFeatureState>(() => {
-    const initialState = {
-      isEnabled: false,
-      variables: {},
-      clientReady: isServerSide ? true : false,
-      didTimeout: false,
-    };
-    if (isServerSide) {
-      return { ...initialState, ...getCurrentValues() };
+  const isClientReady = isServerSide || optimizely.isReady();
+  const decisionStateAndSetter = useState<FeatureDecisionValues>(() => {
+    if (isClientReady) {
+      return getCurrentDecision();
     }
-    return initialState;
+    return { isEnabled: false, variables: {} };
   });
+  let decisionState = decisionStateAndSetter[0];
```

Comment:
Is it safe to mutate the state value from useState?

Reply:
Not sure, I will try removing this.